lib/keywords.cpp

```diff
@@ -0,0 +1,220 @@
+
+#define C11_KEYWORDS \
+    "_Alignas", "_Alignof", "_Atomic", "_Generic", "_Noreturn", "_Static_assert", "_Thread_local"
+
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wunused-macros"
+#endif
+
+#define C23_KEYWORDS \
+    "alignas", "alignof", "bool", "false", "nullptr", "static_assert", "thread_local", "true", "typeof", "typeof_unqual", \
+    "_BitInt", "_Decimal128", "_Decimal32", "_Decimal64"
+
+#ifdef __clang__
+#pragma clang diagnostic pop
+#endif
+
+static const std::unordered_set<std::string> c89_keywords_all = {
+    C90_KEYWORDS
+};
+
+static const std::unordered_set<std::string> c89_keywords = c89_keywords_all;
+
+static const std::unordered_set<std::string> c99_keywords_all = {
+    C90_KEYWORDS, C99_KEYWORDS
+};
+
+static const std::unordered_set<std::string> c99_keywords = {
+    C99_KEYWORDS
+};
+
+static const std::unordered_set<std::string> c11_keywords_all = {
+    C90_KEYWORDS, C99_KEYWORDS, C11_KEYWORDS
+};
+
+static const std::unordered_set<std::string> c11_keywords = {
+    C11_KEYWORDS
+};
+
+/*
+   static const std::unordered_set<std::string> c23_keywords_all = {
+    C90_KEYWORDS, C99_KEYWORDS, C11_KEYWORDS, C23_KEYWORDS
+   };
+
+   static const std::unordered_set<std::string> c23_keywords = {
+    C23_KEYWORDS
+   };
+ */
+
+// see https://en.cppreference.com/w/cpp/keyword
+
+#define CPP03_KEYWORDS \
+    "and", "and_eq", "asm", "auto", "bitand", "bitor", "bool", "break", "case", "catch", "char", \
+    "class", "compl", "const", "const_cast", "continue", "default", \
+    "delete", "do", "double", "dynamic_cast", "else", "enum", "explicit", "export", "extern", "false", \
+    "float", "for", "friend", "goto", "if", "inline", "int", "long", \
+    "mutable", "namespace", "new", "not", "not_eq",  "operator", \
+    "or", "or_eq", "private", "protected", "public", "register", "reinterpret_cast", \
+    "static_cast", "struct", "switch", "template", "this", "throw", \
+    "true", "try", "typedef", "typeid", "typename", "union", "unsigned", "using", \
+    "virtual", "void", "volatile", "wchar_t", "while", "xor", "xor_eq"
+
+#define CPP11_KEYWORDS \
+    "alignas", "alignof", "char16_t", "char32_t", "constexpr", "decltype", \
+    "noexcept", "nullptr", "static_assert", "thread_local"
+
+#define CPP20_KEYWORDS \
+    "concept", "consteval", "constinit", "co_await", \
+    "co_return", "co_yield", "requires"
+
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wunused-macros"
+#endif
+
+#define CPP_TMTS_KEYWORDS \
+    "atomic_cancel", "atomic_commit", "atomic_noexcept", "synchronized"
+
+#define CPP_REFL_TS_KEYWORDS \
+    "reflexpr"
+
+#ifdef __clang__
+#pragma clang diagnostic pop
+#endif
+
+static const std::unordered_set<std::string> cpp03_keywords_all = {
+    CPP03_KEYWORDS
+};
+
+static const std::unordered_set<std::string> cpp03_keywords = cpp03_keywords_all;
+
+static const std::unordered_set<std::string> cpp11_keywords_all = {
```

Comment:
I prefer that the `=` is skipped. 

    static const std::unordered_set<std::string> cpp11_keywords = {
        CPP11_KEYWORDS
    };

=>

    static const std::unordered_set<std::string> cpp11_keywords{
        CPP11_KEYWORDS
    };

Reply:
With the first approach.. the type of RHS is just "generic initializer list" until you look at LHS.

Reply:
As I mentioned in another PR before that won't compile with GCC 4.8.